services/httpd/handler.go

```diff
@@ -461,7 +461,7 @@ func (h *Handler) serveWriteLine(w http.ResponseWriter, r *http.Request, body []
 			}
 
 			// check that the byte is in the standard ascii code range
-			if body[i] > 32 {
+			if body[i] > 32 || i >= len(body)-1 {
```

Comment:
OK, I see what you are doing. Basically, it it's not JSON by index (len-1), it's not JSON at all. Is that what you're getting at?